cerberus/models/booking.py

```diff
@@ -57,38 +69,60 @@ def round_date_time(dt: datetime) -> datetime:
 
         return make_aware(dt)
 
-    def _valid_dates(self) -> bool:
-        return self.end > self.start
-
     def get_overlapping(self) -> QuerySet[Self]:
         start = Q(start__lt=self.start, end__gt=self.start)
         end = Q(start__lt=self.end, end__gt=self.end)
         equal = Q(start=self.start, end=self.end)
+        contains = Q(start__gt=self.start, end__lt=self.end)
 
-        return self.__class__.objects.filter(start | end | equal).exclude(pk=self.pk)  # type: ignore
+        return self.__class__.objects.filter(start | end | equal | contains).exclude(pk=self.pk)
 
     def overlaps(self) -> bool:
         others = self.get_overlapping()
 
         return any(o.bookings.all().count() > 0 for o in others)
 
     def clean(self) -> None:
-        if not self._valid_dates():
-            raise ValidationError("end can not be before start")
-
-        print(f"Checking for other slots between {self.start} and {self.end}")
         if self.overlaps():
             raise ValidationError(f"{self.__class__.__name__} overlaps another {self.__class__.__name__}")
 
-    def move_slot(self, start: datetime, end: datetime | None = None) -> bool:
-        if not all(b.can_move for b in self.bookings.all()):
+        if len({booking.service.id for booking in self.bookings.all()}) > 1:
+            raise ValidationError(f"{self.__class__.__name__} has multiple services")
+
+        if (max_pet := getattr(self.service, "max_pet")) and self.pet_count > max_pet:
+            raise ValidationError(f"{self.__class__.__name__} has max pets for service, {max_pet}")
+
+        if (max_customer := getattr(self.service, "max_customer")) and self.customer_count > max_customer:
+            raise ValidationError(f"{self.__class__.__name__} has max customers for service, {max_customer}")
+
+        super().clean()
+
+    @property
+    def can_move(self) -> bool:
+        return all(b.can_move for b in self.bookings.all())
+
+    def move_slot(self, start: datetime | date, end: datetime | None = None) -> bool:
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (edge_case_not_handled):** Validate the return value of `move_slot` method.

The method always returns `True`, which might not accurately reflect the outcome of the operation. Consider implementing a mechanism to return `False` when the move operation is not successful.

```suggestion
        if not self.can_move:
            return False

        # Your existing logic to move the slot

        # After moving logic
        if self.get_overlapping().exists():  # Assuming get_overlapping() checks for overlaps after moving
            return False  # Move was not successful due to overlaps

        return True  # Move was successful
```